cartridges/int_algolia_sfra/cartridge/static/default/js/algolia/insights-config.js

```diff
@@ -2,56 +2,95 @@
+        } else {
+            // For a single product, the information is at the top level. We recreate the same pidsObj as for a product set.
+            lastPidsObj = [
+                {
+                    pid: data.pid,
+                    qty: data.quantity,
+                },
+            ];
         }
     });
+    // Event defined at https://github.com/SalesforceCommerceCloud/storefront-reference-architecture/blob/dec9c7c684275127338ac3197dfaf8fe656bb2b7/cartridges/app_storefront_base/cartridge/client/default/js/product/base.js#L676
+    // The data in the callback comes from the Cart-AddProduct controller: https://github.com/SalesforceCommerceCloud/storefront-reference-architecture/blob/dec9c7c684275127338ac3197dfaf8fe656bb2b7/cartridges/app_storefront_base/cartridge/controllers/Cart.js#L144
+    $('body').on('product:afterAddToCart', function (event, data) {
+        const objectIDs = [];
+        const objectData = [];
+        const queryID = getUrlParameter('queryID') || lastQueryID;
+        const index = getUrlParameter('indexName') || lastIndexName || productsIndex;
+        let currency;
 
-    // when on search page
-    var searchPage = document.querySelector('.ais-InstantSearch');
-    if (!searchPage) return;
+        lastPidsObj.forEach((pidObj) => {
+            const product = data.cart.items.find((item) => item.id === pidObj.pid);
+            const productInfo = {
+                queryID: queryID,
+                price: product.price.sales.value,
+                quantity: parseInt(pidObj.qty),
+            };
+            if (product.price.list) {
+                // Operation needs to be rounded to avoid "Discount must be a decimal number" errors
+                productInfo.discount = +(
+                    product.price.list.value - product.price.sales.value
+                ).toFixed(2);
+            }
+            currency = product.price.sales.currency;
 
-    var lastQueryID = null;
-    var lastIndexName = null;
+            objectIDs.push(product.id);
+            objectData.push(productInfo);
+        });
 
-    searchPage.addEventListener('click', function (event) {
-        var insightsTarget = findInsightsTarget(event.target, event.currentTarget);
-        if (insightsTarget) {
-            lastQueryID = $(insightsTarget).data('query-id');
-            lastObjectID = $(insightsTarget).data('object-id');
-            lastIndexName = $(insightsTarget).data('index-name');
-        }
-    });
+        const algoliaEventType = queryID
+            ? 'addedToCartObjectIDsAfterSearch'
+            : 'addedToCartObjectIDs';
+        // pliUUID is defined only for a single product add to cart: https://github.com/SalesforceCommerceCloud/storefront-reference-architecture/blob/1d7d4d987d681a11b045746618aec744b4409540/cartridges/app_storefront_base/cartridge/controllers/Cart.js#L125
+        const eventName = data.pliUUID ? 'Product Add to cart' : 'Global Add to cart';
```

Comment:
Is this logic used to distinguish between old and new calls? Sorry, I am just trying to understand logic :)

Reply:
Sure, happy to have your inputs! So the current cartridge sends a `Product Add to cart` event in case of a click on a "Add to Cart" button, and a `Global Add to cart` event in case of a click on a "Add All to Cart" button.
I'm trying to keep a similar logic: send  `Product Add to cart` when a single product is added to the cart and `Global Add to cart` when a product set is added to the cart with the "Add All to Cart" button.

And in the server response, the `pliUUID` field is only set when a single product is added, so I'm using it to determine the event name.